phases/repo.go

```diff
@@ -1,9 +1,120 @@
 package phases
 
-import "fmt"
+import (
+	"fmt"
+	"strings"
 
-// Repo ...
-func Repo() (string, error) {
-	fmt.Println("SetRepo")
-	return "", nil
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/errorutil"
+	"github.com/bitrise-io/go-utils/log"
+)
+
+// RepoDetails encapsulates data needed to perform
+// repo registration related requests through the
+// Bitrise API
+type RepoDetails struct {
+	URL      string
+	Provider string
+	Owner    string
+	Slug     string
+	RepoType string
+}
+
+type urlParts struct {
+	host  string
+	owner string
+	slug  string
+}
+
+func parseURL(cloneURL string) urlParts {
+	parts := strings.SplitAfter(cloneURL, "https://")
```

Comment:
NOTE: we have this tested method: https://github.com/bitrise-steplib/steps-git-clone/blob/master/git.go#L138